decoder/vaapidecoder_h264.cpp

```diff
@@ -1637,6 +1647,19 @@ YamiStatus VaapiDecoderH264::ensureContext(const SharedPtr<SPS>& sps)
             return status;
         // return YAMI_DECODE_FORMAT_CHANGE to info upper layer va context changed
         return YAMI_DECODE_FORMAT_CHANGE;
+    }else if(isResolutionChanged(sps)){
+        m_configBuffer.width = sps->frame_cropping_flag ? sps->m_cropRectWidth
```

Comment:
@xuguangxin , this so many piece code are to reset "m_configBuffer" and "m_videoFormatInfo". "m_configBuffer" is only set here;
"m_videoFormatInfo" is only set in "VaapiDecoderBase::start()", but when just resolution is changed, it will not call "VaapiDecoderBase::start()";
VppInputDecode::read() will access "m_videoFormatInfo" to get new resolution.

I appreciate your advises, more.